ledger/ledger-api-bench-tool/src/test/suite/scala/com/daml/ledger/api/benchtool/submission/NonStakeholderInformeesITSpec.scala

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.benchtool.submission
+
+import com.codahale.metrics.MetricRegistry
+import com.daml.ledger.api.benchtool.config.WorkflowConfig
+import com.daml.ledger.api.benchtool.metrics.MetricsManager.NoOpMetricsManager
+import com.daml.ledger.api.benchtool.services.LedgerApiServices
+import com.daml.ledger.api.testing.utils.SuiteResourceManagementAroundAll
+import com.daml.ledger.api.v1.ledger_offset.LedgerOffset
+import com.daml.ledger.client.binding
+import com.daml.platform.sandbox.fixture.SandboxFixture
+import org.scalatest.{AppendedClues, OptionValues}
+import org.scalatest.flatspec.AsyncFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import scala.concurrent.Future
+
+class NonStakeholderInformeesITSpec
+    extends AsyncFlatSpec
+    with SandboxFixture
+    with SuiteResourceManagementAroundAll
+    with Matchers
+    with AppendedClues
+    with OptionValues {
+
+  it should "divulge events" in {
+    val expectedTemplateNames = Set("Foo1", "Divulger")
+    val submissionConfig = WorkflowConfig.FooSubmissionConfig(
+      numberOfInstances = 100,
+      numberOfObservers = 1,
+      numberOfDivulgees = 3,
+      uniqueParties = false,
+      instanceDistribution = List(
+        WorkflowConfig.FooSubmissionConfig.ContractDescription(
+          template = "Foo1",
+          weight = 1,
+          payloadSizeBytes = 0,
+        )
+      ),
+      nonConsumingExercises = None,
+      consumingExercises = None,
+    )
+    for {
+      ledgerApiServicesF <- LedgerApiServices.forChannel(
+        channel = channel,
+        authorizationHelper = None,
+      )
+      apiServices: LedgerApiServices = ledgerApiServicesF("someUser")
+      submitter = CommandSubmitter(
+        names = new Names(),
+        benchtoolUserServices = apiServices,
+        adminServices = apiServices,
+        metricRegistry = new MetricRegistry,
+        metricsManager = NoOpMetricsManager(),
+      )
+      (signatory, observers, divulgees) <- submitter.prepare(submissionConfig)
+      tested = new FooSubmission(
```

Comment:
submission ?

Reply:
I'm testing `FooSubmission` so I called it `tested`. Does that work?

Reply:
With unit tests I usually call the thing under test `underTest`, this works because it is only a single class.  For in integration tests we testing a lot more then just one class so the naming style doesn't work so well.